>From f22cd9021516ea4080624adafa4806eeeac7fcf8 Mon Sep 17 00:00:00 2001
From: Vincent Legoll <vincent.legoll@gmail.com>
Date: Mon, 11 Aug 2014 08:26:01 +0200
Subject: [PATCH] Updated to also consider llvm-clang for the /usr/lib/ccache
 compiler symlinks
To: vincent.legoll@gmail.com

---
 debian/ccache.triggers.in        |  1 +
 debian/update-ccache-symlinks.in | 15 +++++++++++++++
 2 files changed, 16 insertions(+)

diff --git a/debian/ccache.triggers.in b/debian/ccache.triggers.in
index bd5b1ef..fe25a92 100644
--- a/debian/ccache.triggers.in
+++ b/debian/ccache.triggers.in
@@ -1,2 +1,3 @@
 interest /usr/lib/gcc
 interest /usr/lib/%DEB_HOST_MULTIARCH%/gcc
+interest /usr/lib/llvm*
diff --git a/debian/update-ccache-symlinks.in b/debian/update-ccache-symlinks.in
index 0fbdecf..7662de2 100644
--- a/debian/update-ccache-symlinks.in
+++ b/debian/update-ccache-symlinks.in
@@ -23,6 +23,12 @@ sub consider_gcc {
     consider "${prefix}g++${suffix}";
 }
 
+sub consider_clang {
+    my ($suffix) = @_;
+    consider "clang${suffix}";
+    consider "clang++${suffix}";
+}
+
 # Find existing standard compiler names.
 foreach (@standard_names) {
     consider $_;
@@ -44,6 +50,15 @@ foreach my $dir (<$old_gcc_dir/*>, <$new_gcc_dir/*>) {
     }
 }
 
+# Find existing LLVM-clang variants.
+consider_clang "";
+foreach (</usr/lib/llvm-*>) {
+    if (! -l $_ and -d $_) {
+        s|.*/llvm-||;
+        consider_clang "-$_";
+    }
+}
+
 # Find existing symlinks.
 foreach (<$ccache_dir/*>) {
     if (-l) {
-- 
2.1.0.rc1

